adoc-mode.el

```diff
@@ -1991,6 +1991,20 @@ LANG is a string, and the returned major mode is a symbol."
          (intern (concat lang "-mode"))
          (intern (concat (downcase lang) "-mode")))))
 
+(defun adoc-map-intervals (fun property &optional beg end object)
+  "Apply FUN to all intervals of PROPERTY in OBJECT in the region from BEG to END.
+FUN is called with two arguments: the beginning and the end of the interval."
+  (unless object (setq object (current-buffer)))
+  (unless beg (setq beg (point-min)))
+  (unless end (setq end (point-max)))
+  (let (end-interval)
+    (while
+	(progn
```

Comment:
`progn` is redundant in a `while`.

Reply:
Oh, that's your condition, not the body of the `while`. Might be good to move something in the body and just check for the final state here, as now it looks pretty weird.

Reply:
That is the usual loop with condition at the end.
Example: https://git.savannah.gnu.org/cgit/emacs.git/tree/lisp/jit-lock.el#n324